core/os/memory.h

```diff
@@ -196,6 +197,22 @@ T *memnew_arr_template(size_t p_elements) {
 	return (T *)mem;
 }
 
+// Fast alternative to a loop constructor pattern.
+template <bool p_ensure_zero = false, typename T>
+_FORCE_INLINE_ void memnew_arr_placement(T *p_start, std::size_t p_num) {
+	if constexpr (std::is_trivially_constructible_v<T> && !p_ensure_zero) {
+		// Don't need to do anything :)
+	} else if constexpr (is_zero_constructible_v<T>) {
+		// Can optimize with memset.
+		memset(static_cast<void *>(p_start), 0, p_num * sizeof(T));
+	} else {
+		// Need to use a for loop.
+		for (std::size_t i = 0; i < p_num; i++) {
```

Comment:
I may be a `size_t` defender, but get that `std::` out of my sight
```suggestion
_FORCE_INLINE_ void memnew_arr_placement(T *p_start, size_t p_num) {
	if constexpr (std::is_trivially_constructible_v<T> && !p_ensure_zero) {
		// Don't need to do anything :)
	} else if constexpr (is_zero_constructible_v<T>) {
		// Can optimize with memset.
		memset(static_cast<void *>(p_start), 0, p_num * sizeof(T));
	} else {
		// Need to use a for loop.
		for (size_t i = 0; i < p_num; i++) {
```

Reply:
Oh, that must have been habit from my other project. I found that `std::` is normally required; i think godot gets around it by importing some C header that defines it globally. I removed it!

Reply:
It's required for some types, most notably `std::nullptr_t`; given `nullptr` is only part of C starting in C23, I'd chalk that up to C headers

Granted, even if we used exclusively C++ headers, a good chunk of the compilers don't seem to bother with the `std::` prefix depending on the type. For instance: virtually no compiler will enforce that for the fixed-width ints

Reply:
> a good chunk of the compilers don't seem to bother with the std:: prefix depending on the type.

Yep, that's what threw me off as well. Code compiled fine with `size_t` on my end, but a user reported _their_ compiler (MSVC?) complaining about needing `std::` in front of it.